packages/builder/src/components/start/CreateAppModal.svelte

```diff
@@ -14,20 +14,39 @@
   import { getContext } from "svelte"
   import { fade } from "svelte/transition"
   import { post } from "builderStore/api"
-  import analytics from "../../analytics"
+  import analytics from "analytics"
 
   const { open, close } = getContext("simple-modal")
   //Move this to context="module" once svelte-forms is updated so that it can bind to stores correctly
   const createAppStore = writable({ currentStep: 0, values: {} })
 
   export let hasKey
 
+  let isApiKeyValid
+  let lastApiKey
+  let fetchApiKeyPromise
+  const validateApiKey = async apiKey => {
+    if (!apiKey) return false
+
+    // make sure we only fetch once
+    if (isApiKeyValid === undefined || apiKey !== lastApiKey) {
```

Comment:
Doesn't look like `lastApiKey` gets assigned anywhere here. I also don't think you need all this code here. Pretty sure that something like this should do the same thing:

```js
let currentApiKey

const validateApiKey = async key => { 
  if (key === currentApiKey) return
  currentApiKey = key ? await analytics.identifyByApiKey(key) : false;
  return currentApiKey
}

 apiKey: string()
         .required("Please enter your API key.")
         .test("valid-apikey", "This API key is invalid",  validateApiKey),
```

Reply:
Yes - I did try this, and tried again there to be sure.

There is weirdness that happens with svelte reactivity... where `currentApiKey` doesn't get set immediately, even though you've just set it.... i've seen it before too.

But is causes a race condition. awaiting a shared promise is the only way i could get this to work :/